src/lib.rs

```diff
@@ -410,6 +412,53 @@ impl Contract {
     pub fn get_community(&self, handle: String) -> Option<Community> {
         self.communities.get(&handle)
     }
+
+    pub fn add_featured_community(&mut self, handle: String) {
+        assert!(
+            self.is_moderator(env::predecessor_account_id()),
+            "Only moderators can add featured communities"
+        );
+
+        // Check if the community already exists in the list
+        if self.featured_communities.iter().any(|fc| fc.handle == handle) {
```

Comment:
If set all, there isn't need for check already exists, but we do want to check whether every handle is really a community. You can do that by lookup from `self.communities`